tests/functional/algos/test_algos.py

```diff
@@ -153,16 +152,24 @@ def xfail_if_not_installed(value: dict, import_optional: ImportOptional):
 
 
 class CustomRosenbrock(BenchmarkTask):
-    def __init__(self, max_trials: int = 30, with_fidelity: bool = False):
+    def __init__(
+        self, max_trials: int = 30, with_fidelity: bool = False, seed: int = None
+    ):
         super().__init__(max_trials)
+        self.seed = seed
         self.with_fidelity = with_fidelity
+        self.initialize()
+
+    def initialize(self):
+        self.rng = numpy.random.RandomState(self.seed)
```

Comment:
Oh I see. You're using this method to re-initialize the RNG when reusing the task I think.

Nitpick: I don't remember if the `self.rng` attribute is already type-hinted in the base class. If it isn't, then it's better for the type checker (and also nicer IMO) to declare all attributes in `__init__`.
```suggestion
        self.rng: numpy.random.RandomState
        self.initialize()

    def initialize(self):
        self.rng = numpy.random.RandomState(self.seed)
```